### PR TITLE
IGNITE-19663 Add possibility to check transaction state

### PR DESCRIPTION
**Pull request by sanpwc**
https://issues.apache.org/jira/browse/IGNITE-19663

1. TestOnly IgniteImpl#txManager method added.
2. Assertion added in TxManagerImpl#changeState in order to check that changeState was successfully computed.
3. TestOnly TxManager#pending added.
4. New TxState#PENDING added

### CODE REVIEW

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItCommonApiTest.java [hunk 1]
```diff
@@ -181,7 +181,7 @@ public void checkTimestampOperations() {
 
     /** Check transaction change status with erroneous statements.  */
     @Test
-    public void testTxStateChangedOnErroneousOp() {
+        public void testTxStateChangedOnErroneousOp() {
```
Comment (zstan) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItCommonApiTest.java:184]: redundant

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItCommonApiTest.java [hunk 2]
```diff
@@ -232,7 +232,7 @@ public void testTxStateChangedOnErroneousOp() {
             // No op.
         }
 
-        assertEquals(2, txManagerInternal.finished() - txPrevCnt);
+        assertEquals(4, txManagerInternal.finished() - txPrevCnt);
```
Comment (zstan) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItCommonApiTest.java:235]: plz append pending check here too ?

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItCommonApiTest.java (hunk 1 above)
Comment (sanpwc) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItCommonApiTest.java]: Fixed.

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItCommonApiTest.java (hunk 2 above)
Comment (sanpwc) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItCommonApiTest.java:238]: Added.

**Review by zstan — APPROVED** [review]

**Pull request merged**